examples/step-19/step-19.cc

```diff
@@ -662,21 +666,22 @@ namespace Step19
           for (const auto &particle : particles_in_cell)
             particle_positions.push_back(particle.get_reference_location());
 
-          const Quadrature<dim> quadrature_formula(particle_positions);
-          FEValues<dim>         particle_position_fe_values(mapping,
-                                                    fe,
-                                                    quadrature_formula,
-                                                    update_gradients);
-
-          particle_position_fe_values.reinit(cell);
-
-          // Then we can ask the FEValues object for the gradients of the
-          // solution (i.e., the electric field $\mathbf E$) at these locations
-          // and loop over the individual particles:
-          std::vector<Tensor<1, dim>> field_gradients(
-            quadrature_formula.size());
-          particle_position_fe_values.get_function_gradients(solution,
-                                                             field_gradients);
+          field_values.resize(particle_positions.size());
+          field_gradients.resize(particle_positions.size());
+
+          cell->get_dof_values(solution,
+                               solution_values.begin(),
+                               solution_values.end());
+
+          // Then we can ask the FEPointEvaluation object for the gradients of
+          // the solution (i.e., the electric field $\mathbf E$) at these
+          // locations and loop over the individual particles:
+          FEPointEvaluation<1, dim> evaluator(mapping, fe);
+          evaluator.evaluate(cell,
+                             particle_positions,
+                             solution_values,
```

Comment:
And that would suggest that what this function should take is a `Vector` as third argument.